### PR TITLE
URA-872 - AWS authentication via access keys

### LINKED ISSUE #11: properly handle AWS authentication and update docs

### PR DESCRIPTION
**Pull request by jethror1**
- add ability to authenticate with AWS via access keys defined as environment variables 
  - ability to use SSO/IAM retained through use of `AWS_DEFAULT_PROFILE`
  - closes #11
- adjust tests for new changes
- add github action to add comment to new PRs with reminder to run e2e tests manually
- updated `.secrets.baseline` from removing random strings in `test_upload.py`
- update readme with new `Authentication` section

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/s3_upload/37)

### CODE REVIEW

**Comment by jethror1** [comment]
> ⚠️ Reminder to manually run end to end tests in https://github.com/eastgenomics/s3_upload/tree/main/tests/e2e

All end to end tests passing:
```
jethro@jethro-T490:~/Projects/s3_upload$ python3 -m pytest -v   tests/e2e/
============================================================= test session starts ==============================================================
platform linux -- Python 3.8.10, pytest-7.0.1, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.8.10', 'Platform': 'Linux-5.15.0-124-generic-x86_64-with-glibc2.29', 'Packages': {'pytest': '7.0.1', 'pluggy': '1.0.0'}, 'Plugins': {'Faker': '14.0.0', 'repeat': '0.9.3', 'subtests': '0.11.0', 'html': '4.1.0', 'metadata': '3.0.0', 'mock': '3.11.1', 'cov': '4.0.0', 'random-order': '1.1.1'}}
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
rootdir: /home/jethro/Projects/s3_upload
plugins: Faker-14.0.0, repeat-0.9.3, subtests-0.11.0, html-4.1.0, metadata-3.0.0, mock-3.11.1, cov-4.0.0, random-order-1.1.1
collected 23 items                                                                                                                             

tests/e2e/test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_no_errors_in_logs PASSED                       [  4%]
tests/e2e/test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_remote_files_upload_correctly PASSED           [  8%]
tests/e2e/test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_slack_post_message_after_uploading_as_expected PASSED [ 13%]
tests/e2e/test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_upload_log_file_for_fully_uploaded_run_correct PASSED [ 17%]
tests/e2e/test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_uploaded_run_not_picked_up_to_upload_again PASSED [ 21%]
tests/e2e/test_one_complete_run_upload_to_one_remote_path.py::TestSingleCompleteRun::test_when_nothing_to_upload_that_exit_code_is_zero PASSED [ 26%]
tests/e2e/test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_no_errors_in_logs PASSED                                            [ 30%]
tests/e2e/test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_no_upload_log_generated_for_run_3 PASSED                            [ 34%]
tests/e2e/test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_remote_files_uploaded_correctly PASSED                              [ 39%]
tests/e2e/test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_slack_post_message_after_uploading_as_expected PASSED               [ 43%]
tests/e2e/test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_upload_log_files_for_fully_uploaded_run_1_correct PASSED            [ 47%]
tests/e2e/test_regex_filtering_against_sample_names.py::TestConfigRegexPatternsAgainstSampleNames::test_upload_log_files_for_fully_uploaded_run_2_correct PASSED            [ 52%]
tests/e2e/test_resuming_partially_uploaded_runs.py::TestInterruptAndResume::test_complete_upload_log_file_as_expected PASSED                                                [ 56%]
tests/e2e/test_resuming_partially_uploaded_runs.py::TestInterruptAndResume::test_error_message_for_failed_upload_in_partial_upload_log PASSED                               [ 60%]
tests/e2e/test_resuming_partially_uploaded_runs.py::TestInterruptAndResume::test_partially_uploaded_log_file_as_expected PASSED                                             [ 65%]
tests/e2e/test_resuming_partially_uploaded_runs.py::TestInterruptAndResume::test_slack_post_messages_as_expected PASSED                                                     [ 69%]
tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_no_errors_in_logs PASSED                              [ 73%]
tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_remote_file_uploaded_correctly PASSED                 [ 78%]
tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_slack_post_message_after_uploading_as_expected PASSED [ 82%]
tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_upload_log_files_for_fully_uploaded_run_1_correct PASSED [ 86%]
tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_upload_log_files_for_fully_uploaded_run_2_correct PASSED [ 91%]
tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_uploaded_run_not_picked_up_to_upload_again PASSED     [ 95%]
tests/e2e/test_two_complete_runs_upload_to_two_remote_paths.py::TestTwoCompleteRunsInSeparateMonitorDirectories::test_when_nothing_to_upload_that_exit_code_is_zero PASSED  [100%]

=============================================================================== 23 passed in 10.85s ===============================================================================
```

File: tests/unit/test_log.py [hunk 1]
```diff
@@ -27,14 +27,14 @@ def test_formatter_correctly_set(self):
...
 class TestSetFileHandler(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
-        cls.logger = log.get_logger("s3_upload", log_level=logging.INFO)
-        log.set_file_handler(cls.logger, Path(__file__).parent)
-        cls.logger.setLevel(5)
-
-    @classmethod
-    def tearDownClass(cls):
+    def setUp(self):
```
Comment (fletchercr) [inline tests/unit/test_log.py:30]: I was wondering why you changed from using a class method to instance method here?

Comment (fletchercr) [reply tests/unit/test_log.py:30]: Where are they called in the tests?

**Review by fletchercr — APPROVED** [review]
> Reviewed 9 of 12 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/s3_upload/37)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @jethror1)

**Pull request merged**